src/components/layout/ChatArea.tsx

```diff
@@ -407,11 +429,59 @@ export const ChatArea: React.FC<{
     toggleMemberList,
     openPrivateChat,
     messages,
+    connect,
+    joinChannel,
+    toggleAddServerModal,
   } = useStore();
 
   // Tab completion hook
   const tabCompletion = useTabCompletion();
 
+  const handleIrcLinkClick = (rawUrl: string) => {
+    // Tolerate trailing punctuation in chat text
+    const sanitized = rawUrl.trim().replace(/[),.;:]+$/, "");
+    const urlObj = new URL(sanitized);
+    const host = urlObj.hostname;
+    const scheme = urlObj.protocol.replace(":", "");
+    const port = urlObj.port
+      ? Number.parseInt(urlObj.port, 10)
+      : scheme === "ircs"
+        ? 443
+        : 8000;
+
+    // Channels may be in pathname (/chan1,chan2) or in hash (#chan1,chan2)
+    const rawChannelStr =
+      urlObj.pathname.length > 1
+        ? urlObj.pathname.slice(1)
+        : urlObj.hash.startsWith("#")
+          ? urlObj.hash.slice(1)
+          : "";
+    const channels = rawChannelStr
+      .split(",")
+      .filter(Boolean)
+      .map((c) => decodeURIComponent(c))
+      .map((c) =>
+        c.startsWith("#") ||
+        c.startsWith("&") ||
+        c.startsWith("+") ||
+        c.startsWith("!")
+          ? c
+          : `#${c}`,
+      );
```

Comment:
This adds a lot of business logic for the url parser that should be not part of the ChatArea.tsx, this could be a separated thing that can be tested independently with a few test cases. Specially for cases that are easy and testable like this it is easy to add unit tests.